rclcpp/test/test_time_source.cpp

```diff
@@ -139,7 +139,7 @@ void trigger_clock_changes(
     auto msg = std::make_shared<rosgraph_msgs::msg::Clock>();
     msg->clock.sec = i;
     msg->clock.nanosec = 1000;
-    clock_pub->publish(msg);
+    clock_pub->publish(*msg);
```

Comment:
This is a perfect example of a case where you should use `unique_ptr` instead.

Reply:
Actually, you could use a stack reference instead.

Reply:
Depends on what we think is more overhead, if we know the stack reference will result in a heap allocation (to make space in which to store the copy) then we can just do `unique_ptr`.

Reply:
I used an stack variable here as it is a test with intra process not configured.